### ISSUE TITLE
Callbase not working on latest 4.8 version

### ISSUE THREAD

**Issue by rotem925** [moq/moq4#557]
Hi,
I've upgraded to 4.8.
When setting CallBase=true on new Mock<T>(), and debugging, it seems that the call base does not get called.

Was there any change that could affect this or is there any bug? I didn't see any breaking changes that affect this in the changelog.

Thanks,
Rotem

**Comment by stakx** [comment]
Hi @rotem925 and thanks for reporting.

It's possible that `CallBase` was affected by commit 5837c535df4adc1a3c1e9e897a0462b1ac7a6433. Could you please provide a short code example that demonstrates how `CallBase` no longer works?

**Comment by rotem925** [comment]
Hi @stakx 
Thank you for the quick response,
Here is a simple example of the usage:

```csharp
public class Program
    {
        static void Main(string[] args)
        {
            var mock = new Mock<MyClass>().As<IMyClass>();
            mock.CallBase = true;
            Assert.NotNull(mock.Object.DoSomething()); //this assertion does not pass!!
        }
        public interface IMyClass
        {
            object DoSomething();
        }
        public class MyClass:IMyClass
        {
            public object DoSomething()
            {
                return new object();
            }
        }
    }
```

**Comment by stakx** [comment]
@rotem925 - Thanks a lot for the repro code. It looks like this is a regression introduced by 5837c535df4adc1a3c1e9e897a0462b1ac7a6433:

```diff
-			if (invocation.Method.DeclaringType == typeof(object) || // interface proxy
-				mock.ImplementsInterface(invocation.Method.DeclaringType) && !invocation.Method.LooksLikeEventAttach() && !invocation.Method.LooksLikeEventDetach() && mock.CallBase && !mock.MockedType.GetTypeInfo().IsInterface || // class proxy with explicitly implemented interfaces. The method's declaring type is the interface and the method couldn't be abstract
-				invocation.Method.DeclaringType.GetTypeInfo().IsClass && !invocation.Method.IsAbstract && mock.CallBase // class proxy
-				)
+			if (mock.CallBase && !invocation.Method.IsAbstract)
```

It's a pity this regression wasn't caught earlier. It should be easily fixable, though.

**Comment by rotem925** [comment]
That was fast!
Thank you so much.

**Comment by stakx** [comment]
@rotem925 - The fix for this is included in Moq 4.8.1 which has just been made available on NuGet.

**Comment by rotem925** [comment]
@stakx Thanks for the update!
It works great  now!
